[WIP] Actual Budget Vision (#213)
actualbudget/docs: docs/vision.md

---
title: Actual Budget Vision
---

This document provides an outline of the vision of the open source version of Actual Budget, where the project would like to be, what the goals are and some of the history around Actual Budget. 

### What Makes Actual Budget Great

Some of the things that make Actual Budget stand out from the rest include;

* Local-first

> Actual is local first meaning that the database that powers Actual Budget and ultimatley your budget live on your device, we don't store it on a server, now your wondering what actual server is for, well that allows us to sync changes between budgets, to do this we only send the changes to the server, it stores them and the client pulls down the change(s)

Reviewer: ```suggestion
Actual is local first meaning that the database that powers Actual Budget and ultimatley your budget live on your device. The Actual Budget project doesn't store your information on a centeral server. Now your wondering what actual server is for, well that allows you to sync changes between your budgets from your personal server install. To do this, only changes are sent to the server and ared stored. Your client(s) pull down the change(s)
```

* Open 

> Actual is Open Source, the code is open for anyone to see, view and contribute to. You own your data, you can do whatever you want with it.

Reviewer: ```suggestion
Actual is Open Source utilizing the MIT license, the code is open for anyone to see, view and contribute to. You own your data, you can do whatever you want with it.
```

#### Developer experience

* Move towards component [driven design](https://www.componentdriven.org/) so that a UI testing framework such as [Storybook](https://storybook.js.org/) can be used
* Move towards a component framework scalable across device sizes and aspect ratios, such as [bootstrap](https://getbootstrap.com/)

### Mobile

The current strategy for Actual is to provide a mobile-friendly, responsive website. We do not intend to publish a native app as it is expensive to build and maintain feature parity between the various versions and operating systems. Thus we are fully focusing on a single product.

Instead, Actual Budget will focus on developing an excellent web experience by combining the underlying technology and some UI components from the desktop version with some components maintained separately for mobile and desktop where there need to be significant differences in presentation. For more details, [see our dedicated mobile strategy issue](https://github.com/actualbudget/actual/issues/804).

### The Future

Some of the high-level features we would like to add to Actual Budget in the future include:

* Goal templates moved to the UI.
* Mobile transaction entry.

> * Implementing and testing syncing providers other than Nordigen.

Reviewer: ```suggestion
* Implementing and testing syncing providers in addition to GoCardless.
```

> * Enabling support for Nordigen and other syncing providers by default.

Reviewer: ```suggestion
* Enabling support for additional syncing providers by default.
```

* Darkmode.
* custom reports.

> * and more

Reviewer: ## Third party use
Actual is MIT licensed which allows for use of the software without restriction. The Actual project and community appreciate any submissions from 3rd party contributors to enhance and improve our amazing project. This appreciation extends to any single volunteer or company participants alike.  Let's continue to make Actual Budget the best personal finance tool together!

---
title: Actual Budget Vision
---

> **Vision statement**: Our vision is to be the leading OSS personal finance software used by everyone in any stage of their financial journey. 

Reviewer 2: Suggestion: drop “our vision is” and let the heading speak for itself

**Mission statement**: Our mission is to enable everyone the opportunity to:
* take charge of their finances
* find financial peace of mind
* actively engage in dreaming big and achieving your goals

**Design principles**: Actual is designed using the following principles:
* **Local-first**
* **Open**
* **Sleek and clutter-free**
* **Stable**
* **Reliable**
* **Secure**
* **Privacy focused**
* **Community-driven**

### Local-first

Actual is local first, meaning that the database that powers Actual Budget, and ultimately your budget, lives on your device. We don't store it on a server. Now you're wondering, what 'actual server' is for? Well, that allows you to sync changes to your budget on different devices. To do this, only the changes to the budget are sent to the server. The server stores the messages, and the client pulls down the change(s).

### Open 

Actual is Open Source utilizing the MIT license. The code is open for anyone to see, view, and contribute. You own your data; you can do whatever you want with it.

### User Interface

> 

Reviewer 2: Please don’t interpret this as bringing up the recent “unlock” discussion :) , but I think a note here would be nice along the lines that we make mistakes easy to undo. Similar to the early Macintosh philosophy.
